src/charm.py

```diff
@@ -90,12 +195,49 @@ def _on_get_credentials_action(self, event: ActionEvent) -> None:
             result["postgresql"] = list(postgresql.values())[0]
         if mongodb:
             result["mongodb"] = list(mongodb.values())[0]
+        if kafka:
+            result["kafka"] = list(kafka.values())[0]
         event.set_results(result)
 
     def _on_database_created(self, event: DatabaseCreatedEvent) -> None:
         """Event triggered when a database was created for this application."""
         logger.info(f"database credentials are received: {event.username}")
-        self.unit.status = ActiveStatus(f"received {self.database} credentials")
+        self.unit.status = ActiveStatus(f"Received credentials for database: {self.database}.")
+
+    def _on_topic_created(self, event: TopicCreatedEvent) -> None:
+        """Event triggered when a topic was created for this application."""
+        logger.info(f"Topic credentials are received: {event.username}")
+        self.unit.status = ActiveStatus(f"Received credentials for topic: {self.topic}")
+
+    def _on_database_relation_broken(self, _):
+        """Event triggered when the database relation is removed."""
+        if not self.is_database_related:
+            self.database = ""
+
+    def _on_kafka_relation_broken(self, _):
+        """Event triggered when kafka relatoion is removed."""
+        if not self.is_database_related:
+            self.kafka = ""
```

Comment:
```suggestion
            self.topic = ""
```

Reply:
But really, I'm not sure we should set this to ""

Reply:
I mean, if we unrelate and then relate again, I would expect the same topic/database to be binded.